### ISSUE TITLE
Moving to git-scm.com

### ISSUE THREAD

**Issue by chriscool** [git/git.github.io#729]
In https://lore.kernel.org/git/CAP2yMa+PtRADLB2ZL4U3S8X+87Ddz4UZQNMmtEvV5p9nyoq5SA@mail.gmail.com/ @schacon  said:

> I would love to pull in Git Rev News as a blog on git-scm.com, and/or
> link to Taylor's regular posts about what's new in the new versions.
> It's such great content and would be nice to have more visible.
> Honestly, this whole website would be nice to incorporate:
> https://git.github.io/rev_news/

I am going to reply to his message on the list, but maybe we could also discuss it here. And if we agree on moving to git-scm.com, we could perhaps discuss here how it could happen.

cc @jnareb @mjaix @sivaraam @gitster @stepnem

**Comment by schacon** [comment]
Sorry @chriscool, I didn't actually mean just the Rev News part, I did mean that I thought that all of git.github.io should be part of git-scm.com somehow.

**Comment by schacon** [comment]
Now that git-scm.com is on Hugo, it's probably not difficult to move this content into that tree somewhere, somehow.

**Comment by chriscool** [comment]
No worries @schacon, I am fine with moving all of git.github.io, so that it becomes part of git-scm.com if it's done properly. Let's wait until we hear from the other git.github.io team members I cc'ed above though.

And yeah, I agree it's probably not difficult to move the content. The main issue for me though is that I am not a web developer, so even if it's not difficult to move the content, I am not sure about some details like how to best put redirects in place so that users who bookmarked pages on git.github.io get properly redirected on the corresponding page on git-scm.com, or if there is a need and tools to somehow reformat some pages a bit.

The git.github.io team members would need to get proper credentials to be able to work with git-scm.com as they did with git-scm.com. It might also require a bit of work to replace links that contains git.github.io. (Not sure if we have a lot of relative links by the way.) About the issues and perhaps PRs in https://github.com/git/git.github.io/issues we would need to check if there are some that are worth keeping and then move them over. Also I think it would be nice if the Git history of git.github.io could be kept.

**Comment by To1ne** [comment]
@chriscool I'm really digging this idea, and I'm happy to help. 

> Now that git-scm.com is on Hugo, it's probably not difficult to move this content into that tree somewhere, somehow.

Yeah, I suggest we create a branch that moves the markdown files at their Hugo location and remove all files Hugo isn't using. Then we can merge that branch into git-scm.com.git.

In another branch we should work on the redirects. Here we take the existing markdown files, but in the frontmatter we add something like:

```erb
---
layout: default
title: General Application Information
redirect_to: 'https://git-scm.com/GSoC/General-Application-Information'
---
```

The rest of the file can be deleted.

---

@chriscool The main question for me remains where we want to move pages to. I mean at which locations shall they exist on git-scm.com? I think something like `git-scm.com/rev_news` is pretty good, but there's no real structure in all the other pages.

**Comment by chriscool** [comment]
Thanks @To1ne for volunteering to help on this and for your suggestions! 

I think we need first to agree with people managing git-scm.com on where git.github.io content is going to be moved. Maybe we should actually create an issue on https://github.com/git/git-scm.com/issues and link here to this new issue? Or maybe use the mailing list thread with everyone related to git-scm.org and git.github.io in Cc to get everyone on the same page?

**Comment by chriscool** [comment]
If I could choose, I would prefer we move all the content from git.github.io into `git-scm.com/community/`.

**Comment by sivaraam** [comment]
> Let's wait until we hear from the other git.github.io team members I cc'ed above though.

Sounds fine to me :-)

It would indeed be great to have this site within git-scm.com. We certainly need to work on redirects after tbe migration, though.

> If I could choose, I would prefer we move all the content from git.github.io into `git-scm.com/community/`.

+1

**Comment by chriscool** [comment]
@sivaraam , thanks for your opinion on this.

@jnareb, @mjaix and @stepnem , I would be happy to get your opinion on this too.

**Comment by stepnem** [comment]
> Let's wait until we hear from the other git.github.io team members I cc'ed above though.

I admit that the reasons for migration given so far (better
visibility, 1 site simpler to administrate than 2 sites)
seem neither obvious nor compelling to me (put against
simple inertia of "don't touch what ain't broken", people
being used to the existing location and workflow etc.), but
I don't consider my voice here equal to the others (by a
long shot); I'm only replying to prevent any danger of
becoming a roadblock given the above "until we hear", so
feel free to take this as an OK.

I intend to try to continue my meager contributions no
matter the result of this project.

**Comment by mjaix** [comment]
Hi,

I wholeheartedly agree ***@***.*** [1]'s statements - 
given the past experience of a "curated editing" , I very much enjoyed
the teamwork on polishing the contents
in terms of spelling/phrasing and also sometimes correctness.
Nobody is perfect, and the joy to see the teamwork for each edition
reflects to some extent the spirit of the Git community.
As long as we can keep that spirit, any site or platform change which
results in less hassle for Christian, Jakub or others
is regarded positive from my side.

Best regards,
Markus

[1] @stepnem https://github.com/stepnem

**Comment by sivaraam** [comment]
Would it help any if we continue to maintian the issues / discussion here even if we move the publishing to git-scm.com ? Just trying to understand if that would possibly help address the concern in some way.

**Comment by stepnem** [comment]
> Would it help any if we continue to maintian the issues / discussion here even if we move the publishing to git-scm.com ? Just trying to understand if that would possibly help address the concern in some way.

FWIW, I don't think this alone would be much of an issue (as
opposed to, say, the actual contribution workflow, as
observed e.g. in
https://github.com/git/git-scm.com/pull/1890 (I have no
experience working on git-scm.com, but from a quick look
contributing to git.github.io does currently seem lower
barrier: ~10MB repo vs ~1GB, Markdown & regular git vs Hugo,
scalar, sparse-checkout, apparently long build times)),
i.e., if the consensus is to migrate, it would seem better
to migrate everything (and start building the _new_ inertia,
to use my earlier term).

Bottom line, I agree with @mjaix that what counts (IMHO) is
the convenience of those who do most of the work and
maintenance.  After all, despite the very low barrier to
entry and welcoming attitude of git.github.io, non-team PRs
remain rare, and presumably there will always be the option
of going via one of the team members or just reporting an
issue instead of working on a PR directly.

My earlier comment was only an attempt to provide an honest
personal opinion given my limited involvement.

**Comment by dscho** [comment]
We could have both worlds by pulling in the content from this here repository into git-scm.com, much like the latter already pulls in the ProGit book and its many translations as well as Git's manual pages.

On the other hand, the format does not quite strike me targeting mainly the audience of git-scm.com: The developer spotlight and the featured technical discussion from the Git mailing list would likely be off-putting for the majority of Git users.

Ideally, I would say, the `edition-<nr>.md` file should be split up into the three sections (featured discussion, developer spotlight, and links) and the _links only_ would then be featured on git-scm.com, potentially adding goodness like an RSS feed interested readers can subscribe to.
